src/YProperty.cc

```diff
@@ -50,7 +50,31 @@ YPropertyValue::~YPropertyValue()
 {
 }
 
+bool YPropertyValue::operator==( const YPropertyValue &other ) const
+{
+    // compare the type first
+    if (_type != other.type()) return false;
+
+    // then compare the values
+    switch ( _type )
+    {
+    case YStringProperty:		return _stringVal == other.stringVal();
+    case YBoolProperty:         return _boolVal == other.boolVal();
+    case YIntegerProperty:		return _integerVal == other.integerVal();
+
+    case YUnknownPropertyType:
+    case YOtherProperty:
+        YUI_THROW( YUIException( "Cannot compare " + typeAsStr() + " properties") );
+    }
 
+    // FIXME: never reached, just make the compiler happy (can it be improved?)
```

Comment:
So what happens if you omit it? Ah, I think I know:

If we used a plain `throw` then the compiler would detect that this point is never reached, but `YUI_THROW` is a macro that expands to a [`_YUI_THROW`](https://github.com/libyui/libyui/blob/0ad801d0fcf642a5ac736b9659188ce601cac50f/src/YUIException.h#L927) function. That function should have an `__attribute__((noreturn))` attribute. Try adding it. Hopefully it is possible because it is also a template function :-/

In case it does not help, replace this return with [`__builtin_unreachable ();`](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html#index-g_t_005f_005fbuiltin_005funreachable-4263)